File: perception_cpp/track.cpp
#include "track.hpp"

Track::Track(
    int track_id,
    const std::array<float, 4>& bbox,
    int class_id,
    float confidence
)
    : track_id(track_id),
      bbox(bbox),
      class_id(class_id),
      confidence(confidence),
      kalman()
{
}

File: perception_cpp/python/track.py
from kalman_filter import KalmanFilter

class Track:

    def __init__(
        self,
        track_id,
        bbox,
        class_id,
        confidence
    ):
        self.track_id = track_id
        self.bbox = bbox
        self.class_id = class_id
        self.confidence = confidence

        self.kalman = KalmanFilter()